File: 2. Add Two Numbers/answer.js
// https://leetcode.com/problems/add-two-numbers/
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */

 const getNum = (l) => {
  if (!l) {
      return '';
  }
  return `${getNum(l.next)}${l.val}`;
}

const numToNode = (n) => {
  if (n.length <= 0) {
      return null;
  }
  return new ListNode(parseInt(n.slice(-1)), numToNode(n.slice(0, -1)))
};

const addTwoNumbers = function(l1, l2) {
  const sum = BigInt(getNum(l1)) + BigInt(getNum(l2));
  return numToNode(sum.toString());
};

File: 167. Two Sum II - Input array is sorted/answer.swift
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
class Solution {
  func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
      var tb: [Int: Int] = [:]
      for i in 0..<numbers.count {
          tb[numbers[i]] = i
      }
      for i in 0..<numbers.count {
          if let j = tb[target - numbers[i]], j != i {
              return [i+1, j+1]
          }
      }
      return [-1, -1]
  }
}
